homeassistant/components/iotawatt/config_flow.py

```diff
@@ -0,0 +1,100 @@
+"""Config flow for iotawatt integration."""
+import json
+import logging
+
+import httpx
+from httpx import AsyncClient
+from iotawattpy.iotawatt import Iotawatt
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): str,
+        vol.Required(CONF_HOST): str,
+    }
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    session = AsyncClient()
+    iotawatt = Iotawatt(data["name"], data["host"], session)
+    try:
+        is_connected = await iotawatt.connect()
+    except (KeyError, json.JSONDecodeError, httpx.HTTPError) as err:
```

Comment:
What can raise a `KeyError` ?

Reply:
@balloob The ```connect()``` method from the API can raise a ```KeyError```
```
            try:
                self._macAddress = jsonResults["wifi"]["mac"].replace(":", "")
                LOGGER.debug("MAC: %s", self._macAddress)
            except KeyError:
                raise
```